api/consul.go

```diff
@@ -257,6 +264,61 @@ func (cu *ConsulUpstream) Canonicalize() {
 	}
 }
 
+// ConsulTransparentProxy is used to configure the Envoy sidecar for
+// "transparent proxying", which creates IP tables rules inside the network
+// namespace to ensure traffic flows thru the Envoy proxy
+type ConsulTransparentProxy struct {
+	// UID of the Envoy proxy. Defaults to the default Envoy proxy container
+	// image user.
```

Comment:
The _UID_ is set to the linux user of the container image?

Reply:
Yeah. The `consul-cni` plugin creates an exception to the iptables redirect rules based on UID, so that the Envoy proxy doesn't have all its outbound traffic redirected to itself in a loop. (Yes, that allows a task with the same UID to bypass Envoy... but that's how it transparent proxy works with Consul + K8s too.)